tests/ui/job-view/Filtering_test.jsx

```diff
@@ -0,0 +1,275 @@
+          author: 'whozat@gmail.com',
+          revision_count: 1,
+          push_timestamp: 1562867957,
+          repository_id: 4,
+          revisions: [
+            {
+              repository_id: 4,
+              revision: '3333333333335143b8df3f4b3e9b504dfbc589a0',
+              author: 'whozat <whozat@gmail.com>',
+              comments: 'didathing',
+            },
+          ],
+        },
+      ],
+    });
+    fetchMock.get(`begin:${getApiUrl('/jobs/')}`, jobListFixtureOne);
+  });
+
+  afterEach(() => {
+    cleanup();
+    replaceLocation({});
+  });
+
+  const jobCount = () => document.querySelectorAll('.job-btn').length;
+
+  describe('by author', () => {
+    beforeEach(() => {
+      fetchMock.get(
+        getProjectUrl('/push/?full=true&count=10&author=reviewbot', 'autoland'),
+        {
+          results: [
+            {
+              id: 111111,
+              revision: '3333333333335143b8df3f4b3e9b504dfbc589a0',
+              author: 'reviewbot',
+              revision_count: 1,
+              push_timestamp: 1562867957,
+              repository_id: 4,
+              revisions: [
+                {
+                  repository_id: 4,
+                  revision: '3333333333335143b8df3f4b3e9b504dfbc589a0',
+                  author: 'reviewbot <reviewbot>',
+                  comments: 'didathing',
+                },
+              ],
+            },
+          ],
+        },
+      );
+    });
+
+    test('should have 1 push', async () => {
+      const { getAllByText, getByTestId } = render(<App />);
+      await waitForElement(() => getAllByText('reviewbot'));
+      setUrlParam('author', 'reviewbot');
+      await waitForElementToBeRemoved(() => getByTestId('push-511138'));
+
+      const filteredPushes = await waitForElement(() =>
+        getAllByText('View Tests'),
+      );
+      expect(filteredPushes).toHaveLength(1);
+
+      setUrlParam('author', null);
+      await waitForElement(() => getAllByText('jarilvalenciano@gmail.com'));
+      const unFilteredPushes = await waitForElement(() =>
+        getAllByText('View Tests'),
+      );
+      expect(unFilteredPushes).toHaveLength(10);
+    });
+  });
+
+  describe('by failure result', () => {
+    beforeEach(() => {
+      // polling..  Return nothing...
+      fetchMock.get(
+        `begin:${getProjectUrl(
+          '/push/?full=true&count=11&push_timestamp__lte=',
+          'autoland',
+        )}`,
+        {
+          results: [],
+        },
+      );
+    });
+
+    test('should have 20 failures', async () => {
+      const { getAllByText, getByTitle, findAllByText } = render(<App />);
+      await waitForElement(() => findAllByText('B'));
+      const unclassifiedOnlyButton = getByTitle(
+        'Loaded failures / toggle filtering for unclassified failures',
+      );
+      fireEvent.click(unclassifiedOnlyButton);
+
+      await waitForElementToBeRemoved(() => getAllByText('yaml'));
```

Comment:
Correct.  Since ``yaml`` is not an unclassified failure, making this call will ensure that the filtering has completed.  Then we can get an accurate count of what's left.